File: ExampleApp/app/src/main/java/com/motiv/example/PhotosPagerAdapter.java
package com.motiv.example;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.LinearLayout;
import androidx.viewpager.widget.PagerAdapter;
import com.motiv.example.databinding.PhotospageradapterBinding;
import com.squareup.picasso.Picasso;
import java.util.ArrayList;

public class PhotosPagerAdapter extends PagerAdapter {

    private ArrayList<Photo> data = new ArrayList();
    private LinearLayout linearlayout00;
    private ImageView imageview10;

    @Override
    public int getCount() {
        return data.size();
    }

    @Override
    public boolean isViewFromObject(View view, Object o) {

        return view == o;
    }

    public void setData(java.util.List<com.motiv.example.Photo> newData) {
        data.clear();
        data.addAll(newData);
        notifyDataSetChanged();
    }

    @Override
    public void destroyItem(ViewGroup container, int position, Object o) {
        container.removeView((View) o);
    }

    @Override
    public Object instantiateItem(ViewGroup container, int position) {
        LayoutInflater inflater = LayoutInflater.from(container.getContext());
        PhotospageradapterBinding binding = PhotospageradapterBinding.inflate(inflater);
        Photo photo = data.get(position);

        Picasso.with(container.getContext()).load(photo.getUrl()).into(binding.imageview10);
        ;

        container.addView(binding.getRoot());
        return binding.getRoot();
    }
}

File: ExampleApp/app/src/main/java/com/motiv/example/PhotosPagerAdapter.kt
 
package com.motiv.example
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.LinearLayout
import androidx.viewpager.widget.PagerAdapter
import com.squareup.picasso.Picasso
import kotlin.collections.List

public class PhotosPagerAdapter : PagerAdapter() {

    private var data: ArrayList<Photo> = ArrayList()

    private lateinit var linearlayout00: LinearLayout

    private lateinit var imageview10: ImageView

    override fun getCount(): Int {
        return data.size
    } override fun isViewFromObject(view: View, o: Any): Boolean {
        return view == o
    } fun setData(newData: List<com.motiv.example.Photo>) {
        data.clear()
        data.addAll(newData)
        notifyDataSetChanged()
    } override fun destroyItem(container: ViewGroup, position: Int, o: Any) {
        container.removeView(o as View)
    } override fun instantiateItem(container: ViewGroup, position: Int): Any {
        val inflater = LayoutInflater.from(container.getContext())
        val v = inflater.inflate(R.layout.photospageradapter, container, false)
        linearlayout00 = v.findViewById<LinearLayout>(R.id.linearlayout00)
        imageview10 = v.findViewById<ImageView>(R.id.imageview10)

        val photo: Photo = data.get(position)

        Picasso.with(container.getContext()).load(photo.getUrl()).into(imageview10)

        container.addView(v)
        return v
    }
}
